src/gui/properties/speedwidget.h

```diff
@@ -0,0 +1,81 @@
+#ifndef SPEEDWIDGET_H
+#define SPEEDWIDGET_H
+
+#include <QWidget>
+#include <QVBoxLayout>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QComboBox>
+#include <QtConcurrentRun>
+#include <QToolButton>
+#include <QMenu>
+#include <QSignalMapper>
+
+#include "speedplotview.h"
+
+class PropertiesWidget;
+
+class SpeedWidget : public QWidget
+{
+    Q_OBJECT
+public:
+    SpeedWidget(PropertiesWidget *parent);
+    ~SpeedWidget();
+
+    void update();
+
+signals:
+    void speedChanged();
+
+public slots:
+    void graphUpdate();
+    void on_period_change(int period);
+    void on_graph_change(int id);
+    void loadSettings();
+    void saveSettings() const;
+
+private:
+    QVBoxLayout *m_layout;
+    QHBoxLayout *m_hlayout;
+    QLabel *m_period_label;
+    QComboBox *m_period_combobox;
+    QSpeedPlotView *m_plot;
+    PropertiesWidget *m_properties;
+
+    QToolButton *m_graphs_button;
+    QMenu *m_graphs_menu;
+    QList<QAction *> graphs_menu_actions;
+    QSignalMapper *m_graphs_signal_mapper;
+
+    enum TimePeriod {
```

Comment:
Move open brace to next line. See [Coding style](https://github.com/qbittorrent/qBittorrent/wiki/Coding-style).